### ISSUE TITLE
Regression issues with final modifiers in replaceable class definitions

### ISSUE THREAD

**Issue by abuntrock** [OpenModelica/OpenModelica#11964]
### Description

I discovered some problems with the handling of final modifications in replaceable class defintions in OMEdit 1.22.1 (and in todays nightly v1.23.0-dev-297-g308aba5ec2), which were already fixed (or not yet present) in OMEdit 1.22.0.

### Steps to Reproduce

1. Take the following modelica package:
```modelica
package FinalModifierReplaceable
    model ClassWithInstance
     TopLevelClass topLevelClass
     annotation(Placement(visible = true, transformation(origin = {0, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
  end ClassWithInstance;

  model TopLevelClass
    parameter Real realParam = 5;
    parameter Boolean boolParam1 = false;
    parameter Boolean boolParam2 = false annotation (choices(checkBox=true));
     replaceable FinalModifierReplaceable.SubLevel1 replModel(
     final realParam = realParam,
     final boolParam1 = boolParam1,
     final boolParam2 = boolParam2)
    annotation(choices(choice( redeclare FinalModifierReplaceable.SubLevel1 replModel "Some description")));
  end TopLevelClass;

  model SubLevel1
    parameter Real realParam = 5;
    parameter Boolean boolParam1 = false;
    parameter Boolean boolParam2 = false annotation (choices(checkBox=true));
    parameter Real sublevelParameter = 5;
  end SubLevel1;
end FinalModifierReplaceable;
```
2. Open the class `ClassWithInstance` and open the parameter dialog of `topLevelClass`.
3. Open the replaceable subdialog `topLevelClass.replModel` via the edit button.  Here the first problem appears. All the parameters which are set to final in the replaceable defintion of `replModel` in `TopLevelClass`. This is the same problem as reported here #10394, which was fixed for version 1.22.0.
4. Close the replaceable subdialog via the "OK" button again. The following error appears:
`[FinalModifierReplaceable: 12:128-12:151]: [Trying to override final element boolParam2 with modifier '= false'.](omeditmessagesbrowser:///FinalModifierReplaceable?lineNumber=12)`
The text layer of of `ClassWithInstance` now looks like this:
```modelica
model ClassWithInstance
 TopLevelClass topLevelClass(replModel(final boolParam2 = false))
 annotation(Placement(visible = true, transformation(origin = {0, 0}, extent = {{-10, -10}, {10, 10}}, rotation = 0)));
end ClassWithInstance;
```
5. Somehow the value of `boolParam2` is overwritten. It seems to depend on the existence of the `choices(checkBox=true)` annotation because the value of `boolParam1` is not overwritten.

### Expected Behavior

1. Final parameters are not shown in the replaceable sub-dialog as already fixed for 1.22.0 in #10394.
2. Boolean parameters with a checkbox annotation are not overwritten by opening the replaceable class sub-dialog. 

### Version and OS

 - OpenModelica Version: 1.22.1 and v1.23.0-dev-297-g308aba5ec2 (nightly from 8.2.2024)
 - OS: Windows 10, 64 bit

This is really important for us because it makes a lot of out parameterized models unusable in OMEdit 1.22.1.
Thanks for looking into this
Aaron Buntrock - Bosch Rexroth

**Comment by casella** [comment]
@abuntrock we are going to release 1.22.2 because it has some critical fixes. We'll make a 1.22.3 release in 2-3 weeks addressing this issue.

**Comment by adeas31** [comment]
@abuntrock PR #12022 fixes this issue. Test this with tomorrow's nightly build.

**Comment by abuntrock** [comment]
I tested this with todays nightly and it looks like expected and as it was in 1.22.0. Thank you very much. We would be very happy to see this in a possible 1.22.3 version to combine the performance advantages made for 1.22.2 with the correct look of the subdialog.

**Comment by casella** [comment]
> I tested this with todays nightly and it looks like expected and as it was in 1.22.0. Thank you very much. We would be very happy to see this in a possible 1.22.3 version to combine the performance advantages made for 1.22.2 with the correct look of the subdialog.

No problem, @adeas31 can you please port #12022 to maintenance/v1.22?

**Comment by adeas31** [comment]
Fixed with #12022 and ported to maintenance/v1.22 with #12026
